src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -530,6 +545,39 @@ private void sendCitizenPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasN
         }
     }
 
+    /**
+     * Sends packages to update the workOrders.
+     * @param oldSubscribers the existing subscribers.
+     * @param hasNewSubscribers the new subscribers.
+     */
+    private void sendWorkOrderPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
+    {
+        if (getWorkManager().isDirty() || hasNewSubscribers)
+        {
+            int i = 0;
+            for (AbstractWorkOrder workOrder : getWorkManager().getWorkOrders().values())
+            {
+
+                ColonyViewWorkOrderMessage msg = new ColonyViewWorkOrderMessage(this, workOrder , i++ , ColonyViewWorkOrderMessage.workOrderMessages.EDIT);
+                subscribers.stream().filter(player -> workManager.isDirty() || !oldSubscribers.contains(player))
+                           .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
+            }
+
+            if(getWorkManager().getWorkOrders().isEmpty())
+            {
+                ColonyViewWorkOrderMessage msg = new ColonyViewWorkOrderMessage(this, new WorkOrderBuild() , 0 , ColonyViewWorkOrderMessage.workOrderMessages.EMPTY);
```

Comment:
Why is this needed? We send a special packet when we have no work orders? If this is something we need, it seems like what we really need is an individual delete.